gensim/models/word2vec.py

```diff
@@ -1610,13 +1612,16 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar, u
                     logger.debug("skipping line #%i with OOV words: %s" % (line_no, line.strip()))
                     continue
 
+                original_vocab = self.vocab
+                self.vocab = ok_vocab
                 ignore = set(self.vocab[v].index for v in [a, b, c])  # indexes of words to ignore
                 predicted = None
                 # find the most likely prediction, ignoring OOV words and input words
                 sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
```

Comment:
Yes, I agree it's currently a subtle bug that this call to `most_similar()` passes the `restrict_vocab` argument, since it may not mean the exact same thing. (It's not always the case that the word-vectors have already been frequency-sorted.) It's only truly correct to pass the count in if the interpretation is the same. 

But, it's likely a big speedup, in cases where the vocabulary is large compared to the `restrict_vocab`, as it prevents lots of unnecessary distance-calculations in the `dot(...)` line of `most_similar()`. (Executing `most_similar()` with a filtered `vocab` doesn't provide the same optimization.) 

Perhaps we just want to document that for `restrict_vocab` to work as expected here, the vocabulary must be sorted? (Or even redefine the argument here to mean limiting searches to the 1st indexes, whether they're the most-frequent or not. So no extra sort-by-frequency here.)